tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTests.java

```diff
@@ -462,6 +467,137 @@ public void noLRATest() throws WebApplicationException {
+        joinWithOneResource("joinWithOneResourceSameMethodTwiceWithCancel",
+                false, JOIN_WITH_EXISTNG_LRA_PATH, JOIN_WITH_EXISTNG_LRA_PATH);
+    }
+
+    /**
+     * client invokes the same participant twice but using different methods in the same LRA context
+     * cancel the LRA
+     * check that the participant was asked to compensate only once
+     */
+    @Test
+    public void joinWithOneResourceDifferentMethodTwiceWithCancel() throws WebApplicationException {
+        joinWithOneResource("joinWithOneResourceDifferentMethodTwiceWithCancel",
+                false, JOIN_WITH_EXISTNG_LRA_PATH, JOIN_WITH_EXISTNG_LRA_PATH2);
+    }
+
+    /**
+     * client invokes the same participant method twice in the same LRA context
+     * close the LRA
+     * check that the participant was asked to complete only once
+     */
+    @Test
+    public void joinWithOneResourceSameMethodTwiceWithClose() throws WebApplicationException {
+        joinWithOneResource("joinWithOneResourceSameMethodTwiceWithClose",
+                true, JOIN_WITH_EXISTNG_LRA_PATH, JOIN_WITH_EXISTNG_LRA_PATH);
+    }
+
+    /**
+     * client invokes the same participant twice but using different methods in the same LRA context
+     * close the LRA
+     * check that the participant was asked to complete only once
+     */
+    @Test
+    public void joinWithOneResourceDifferentMethodTwiceWithClose() throws WebApplicationException {
+        joinWithOneResource("joinWithOneResourceDifferentMethodTwiceWithClose",
+                true, JOIN_WITH_EXISTNG_LRA_PATH, JOIN_WITH_EXISTNG_LRA_PATH2);
+    }
+
+    /**
+     * client invokes different participant in the same LRA context
+     * close the LRA
+     * check that both participants were asked to complete
+     */
+    @Test
+    public void joinWithTwoResourcesWithClose() throws WebApplicationException {
+        joinWithTwoResources(true);
+    }
+
+    /**
+     * client invokes different participants in the same LRA context
+     * cancel the LRA
+     * check that both participants were asked to compensate
+     */
+    @Test
+    public void joinWithTwoResourcesWithCancel() throws WebApplicationException {
+        joinWithTwoResources(false);
+    }
+
+    private void joinWithOneResource(String methodName, boolean close, String resouce1Method, String resouce2Method)
+            throws WebApplicationException {
+        // set up the web target for the test
+        WebTarget resource1Path = tckSuiteTarget.path(TCK_PARTICIPANT_RESOURCE_PATH).path(resouce1Method);
+        WebTarget resource2Path = tckSuiteTarget.path(TCK_PARTICIPANT_RESOURCE_PATH).path(resouce2Method);
+
+        URI lra = lraClient.startLRA(null, lraClientId(), lraTimeout(), ChronoUnit.MILLIS);
+
+        // get the initial values for the number of completions and compensations
+        int completedCount = getActivityCount(TCK_PARTICIPANT_RESOURCE_PATH, COMPLETED_CNT_PATH);
```

Comment:
I would put the retrieval of the initial values first before any interaction with Participant/coordincator so that we are absolutely sure about the interactions.